spline-shadow/persistence/src/main/scala/za/co/absa/spline/persistence/DataLineageTransactionParams.scala

```diff
@@ -18,135 +18,114 @@ package za.co.absa.spline.persistence
 
 import java.lang.Iterable
 import java.util.UUID
+import java.util.UUID.randomUUID
 
-import scala.collection.JavaConverters._
+import com.arangodb.velocypack.VPackSlice
+import org.apache.commons.lang.builder.ToStringBuilder.reflectionToString
 import za.co.absa.spline.model.{DataLineage, MetaDataset}
+import za.co.absa.spline.persistence.model._
 import za.co.absa.spline.{model => splinemodel}
-import za.co.absa.spline.model.arango._
-import io.circe.ObjectEncoder
-import io.circe.generic.semiauto.deriveEncoder
-import org.apache.commons.lang.builder.ToStringBuilder.reflectionToString
-import java.util.UUID.randomUUID
-// import com.outr.arango.managed._ needed for decoder creation
-import com.outr.arango.managed._
-
-case class DataLineageTransactionParams(
-  operation: Iterable[String],
-  follows: Iterable[String],
-  dataSource: Iterable[String],
-  writesTo: Iterable[String],
-  readsFrom: Iterable[String],
-  app: Iterable[String],
-  implements: Iterable[String],
-  execution: Iterable[String],
-  executes: Iterable[String],
-  progress: Iterable[String],
-  progressOf: Iterable[String]) {
-
-    def fields: Array[String] = getClass
-      .getDeclaredFields
-      .map(_.getName)
-      .filter(_ != "$outer")
-
-    def saveCollectionsJs: String = fields
-      .map(field => s"""
-          |  console.log('start $field');
-          |  params.$field.forEach(o => {
-          |    const doc = JSON.parse(o);
-          |    db.$field.insert(doc);
-          |  });
-          |  console.info('end $field');
+
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+
+case class DataLineageTransactionParams
+(
+  operation: Iterable[VPackSlice],
+  follows: Iterable[VPackSlice],
+  dataSource: Iterable[VPackSlice],
+  writesTo: Iterable[VPackSlice],
+  readsFrom: Iterable[VPackSlice],
+  app: Iterable[VPackSlice],
+  implements: Iterable[VPackSlice],
+  execution: Iterable[VPackSlice],
+  executes: Iterable[VPackSlice],
+  progress: Iterable[VPackSlice],
+  progressOf: Iterable[VPackSlice]) {
+
+  def fields: Array[String] = getClass
+    .getDeclaredFields
+    .map(_.getName)
+    .filter(_ != "$outer")
+
+  def saveCollectionsJs: String = fields
+    .map(field =>
+      s"""
+         |  params.$field.forEach(o => db.$field.insert(o));
```

Comment:
In this case you should be able to do ``` db.$field.insert(params.$field) ```

Reply:
I tried :) but it didn't work. In JS in order to use method as lambda, provided that the method needs to have access to `this`, you'd need to bind the method to the object explicitly.

Reply:
What I mean is that insert method accepts also arrays and not only single values.

Reply:
Actually I am not sure if ```insert``` can do it but ```save``` should be able to.

Reply:
Oh, yes, sorry, I misread your comment (thought it were `...forEach(db.field.insert)`)
Regarding `insert(array)` I tried it as well. The problem with it was that this method doesn't throw exceptions. Instead it returns an array of results per item. But since we need to fail fast and abort the transaction on an error in any item I thought `forEach(o => insert(o))` does better.

Reply:
According to the doc `insert` is just an alias to `save`, so no matter which one to use.